trackintel/preprocessing/trips.py

```diff
@@ -0,0 +1,425 @@
+        The staypoints have to follow the standard definition for staypoints DataFrames. The location ID column
+        is necessary to connect trips via locations to a tour. If None, trips will be connected based only on a
+        distance threshold `max_dist`.
+
+    max_dist: float, default 100 (meters)
+        Maximum distance between the end point of one trip and the start point of the next trip on a tour.
+        Note: If `max_nr_gaps > 0` (see below), a tour can contain gaps
+
+    max_time: Timedelta, default 1 day
+        Maximum time that a tour is allowed to take
+
+    max_nr_gaps: int, default 0
+        Maximum number of spatial gaps on the tour. Use with caution - see notes below.
+
+    print_progress : bool, default False
+        If print_progress is True, the progress bar is displayed
+
+
+    Returns
+    -------
+    trips: GeoDataFrame (as trackintel trips)
+        Same as `trips_inp`, but with additional column `tour_id`. This is the id of the smallest tour that the trip
+        belongs to (see notes below).
+
+    tours: GeoDataFrame (as trackintel tours)
+        The generated tours
+
+    Notes
+    -------
+    - Tours are defined as a collection of trips in a certain time frame that start and end at the same point
+    - Nested tours are possible and will be regarded as 2 (or more tours), where the smaller tour is also part of the
+      bigger tour. Therefore, one trip can belong to multiple tours! In the `tour_id` column of the trips table,
+      only the id of the smallest tour is used.
+    - It is possible to allow spatial gaps to occur on the tour, which might be useful to deal with missing data.
+      Example: The two trips home-work, supermarket-home would still be detected as a tour when max_nr_gaps >= 1,
+      although the work-supermarket trip is missing.
+      Warning: This only counts the number of gaps, but neither temporal or spatial distance of gaps, nor the number
+      of missing trips in a gap are bounded. Thus, this parameter should be set with caution, because trips that are
+      hours apart might still be connected to a tour if `max_nr_gaps > 0`.
+    """
+    # Two options: either the location IDs for staypoints on the trips are provided, or a maximum distance threshold
+    # between end and start of trips is used
+    if staypoints is not None:
+        assert (
+            "location_id" in staypoints.columns
+        ), "Staypoints with location ID is required, otherwise tours are generated without location using max_dist"
+        geom_col = None  # not used
+        crs_is_projected = False  # not used
+    else:
+        # if no location is given, we need the trips table to have a geometry column
+        assert isinstance(trips_inp, gpd.geodataframe.GeoDataFrame), "Trips table must be a GeoDataFrame"
+        geom_col = trips_inp.geometry.name
+        # get crs
+        crs_is_projected = ti.geogr.distances.check_gdf_crs(trips_inp.copy())
+
+    trips = trips_inp.copy()
```

Comment:
According to our new convention, the long and short name of trips are both `trips`. That mean this line will become `trips = trips.copy()`, I am not sure if there will be some problems if we wrote this...

Reply:
See new commit, I renamed both the `trips` as well as the `trips_inp` variable now